buku

```diff
@@ -6080,22 +6090,28 @@ POSITIONAL ARGUMENTS:
             bdb.close_quit(1)
 
         # Parse tags into a comma-separated string
-        tags = DELIM
-        keywords = args.add
-        if tags_in is not None:
-            if tags_in[0] == '+':
-                if len(tags_in) > 1:
-                    # The case: buku -a url tag1, tag2 --tag + tag3, tag4
-                    tags_in = tags_in[1:]
-                    # In case of add, args.add may have URL followed by tags
-                    # Add delimiter as url+tags may not end with one
-                    keywords = args.add + [DELIM] + tags_in
+        # --add may have URL followed by tags
+        keywords_except, keywords = [], args.add[1:]
+        # taglists are taken from --add (starting from 2nd value) and from --tags
+        # if taglist starts with '-', its contents are excluded from resulting tags
+        # if BOTH taglists is are either empty or start with '+'/'-', fetched tags are included
```

Comment:
Ended up changing the logic here a little; I had to think hard what would make sense to do here, so I decided to elaborate in a comment and made sure to test it thoroughly as well.